packages/react-graphql/src/async.tsx

```diff
@@ -47,14 +49,31 @@ export function createAsyncQueryComponent<Data, Variables, DeepPartial>({
     props: Omit<QueryProps<Data, Variables>, 'query'> & ConstantProps,
   ) {
     const [componentProps, asyncProps] = splitProps(props);
+    const client = useApolloClient();
 
     return (
       <Async
         id={id}
         load={load}
         defer={defer}
         render={query =>
-          query ? <Query query={query} {...componentProps as any} /> : null
+          query ? (
+            <Query query={query} {...componentProps as any} />
+          ) : (
+            componentProps.children({
+              data: undefined,
+              loading: true,
+              variables: (componentProps as any).variables,
+              client,
+              networkStatus: NetworkStatus.loading,
+              refetch: noop as any,
+              fetchMore: noop as any,
+              updateQuery: noop as any,
+              startPolling: noop as any,
```

Comment:
my only real concern is that with these functions as no-ops it's possible you could get into weird situations where a component tries to call `startPolling` or `subscribeToMore` immediately and that just fails silently because the initial render / `componentDidMount`  was given a `noop`

Reply:
This is true but unfortunately there isn't anything we can do about that here, hopefully they would correctly update their poll when this function changes.